### PR TITLE
Expose the `geo` module

### PR DESCRIPTION
**Pull request by SupernaviX**
👋 

I'm doing some rust simulation work for Leios, and trying to use `netsim-async` for modeling the network. This library has a useful `geo` module which can compute the latency between two points on Earth, but it isn't exposed in a way that's very easy to use. (The only public function is `EdgePolicy::between_nodes`, which is hard to use and doesn't let you adjust the result.)

This PR makes the `geo` module public in `netsim-core`, and re-exports it through `netsim` and `netsim-async`. It also fixes some clippy errors which were making linting fail.

### CODE REVIEW

File: netsim-core/Cargo.toml [hunk 1]
```diff
@@ -9,3 +9,6 @@ license = "Apache-2.0"
 [dependencies]
 anyhow = "1.0.79"
 logos = "0.14.0"
+
+[features]
+thread_sleep_until = []
```
Comment (SupernaviX) [inline netsim-core/Cargo.toml:14]: This fixes a clippy warning about a `#[cfg(feature = "thread_sleep_until")]` check. There's an unstable rust feature named `thread_sleep_until`, but I don't think you can check for unstable features (or for a nightly rustc version) without adding a build script, and this was easier.

File: netsim-core/src/geo.rs [hunk 2]
```diff
@@ -173,13 +172,7 @@ pub fn latency_between_locations(p1: Location, p2: Location, sol_fo: f64) -> Opt
...
     const SPEED_OF_FIBER: f64 = SPEED_OF_LIGHT * 0.69; // light travels 31% slower in fiber optics
 
-    let sol_fo = if sol_fo < 0.01 {
-        0.01
-    } else if sol_fo > 1.0 {
-        1.0
-    } else {
-        sol_fo
-    };
+    let sol_fo = sol_fo.clamp(0.01, 1.0);
```
Comment (SupernaviX) [inline netsim-core/src/geo.rs:175]: Clippy fix

File: netsim-core/src/policy.rs [hunk 3]
```diff
@@ -92,7 +92,7 @@ impl Latency {
         Self(duration)
     }
 
-    pub(crate) fn to_duration(self) -> Duration {
+    pub fn to_duration(self) -> Duration {
```
Comment (SupernaviX) [inline netsim-core/src/policy.rs:95]: The `geo::latency_between_locations`  function returns this opaque `Latency` struct, and I wanted to add extra latency between some nodes. So I publicly exposed this `to_duration` method so consumers can use the result in computations.

File: netsim-core/src/geo.rs (hunk 2 above)
Comment (NicolasDP) [reply netsim-core/src/geo.rs:175]: nice, easier to read too.

File: netsim-core/Cargo.toml (hunk 1 above)
Comment (NicolasDP) [reply netsim-core/Cargo.toml:14]: that's new. Clippy is quite annoying. I thought adding the `#[cfg()]` check would be future proofing the build/lint and leave a hint of what would make more sense to use in the future once stabilised.

Thanks for adding this and fixing the lint

**Review by NicolasDP — APPROVED** [review]
Thank you for the contribution.

**Pull request merged**